pkg/registry/core/pod/strategy.go

```diff
@@ -292,10 +295,9 @@ var ResizeStrategy = podResizeStrategy{
 
 // dropNonResizeUpdates discards all changes except for pod.Spec.Containers[*].Resources, pod.Spec.InitContainers[*].Resources, ResizePolicy and certain metadata
 func dropNonResizeUpdates(newPod, oldPod *api.Pod) *api.Pod {
-	pod := dropPodUpdates(newPod, oldPod)
-
 	// Containers are not allowed to be re-ordered, but in case they were,
 	// we don't want to corrupt them here. It will get caught in validation.
+	pod := oldPod.DeepCopy()
```

Comment:
nit, name this to make it clear this is a copy of the old pod we're messing with (strawman: `oldPodCopyWithMergedResources`)

Reply:
done!